### ISSUE TITLE
Improve API endpoint auth with scopes

### ISSUE THREAD

**Issue by monotasker** [MESH-Research/knowledge-commons-repository#425]

**Comment by mayel** [comment]
Is this about the API of https://works.hcommons.org ? I was trying to authenticate with OAuth2 and got an `invalid_scopes` error, so wondering if it's the same issue. I used the following config on the client:
```
          base_uri = "https://works.hcommons.org"
          client_id: System.get_env("KCWORKS_CLIENT_ID"),
          client_secret: System.get_env("KCWORKS_CLIENT_SECRET"),
          authorize_uri: "#{base_uri}/oauth/authorize",
          access_token_uri: "#{base_uri}/oauth/token",
          userinfo_uri: "#{base_uri}/api/me",
          grant_type: System.get_env("KCWORKS_GRANT_TYPE", "authorization_code"),
          scope: System.get_env("KCWORKS_SCOPE", "deposit:write deposit:actions"),

**Comment by monotasker** [comment]
Yes, this issue is related to the API validation, but we don't currently have oauth2 authorization available for the API. If you need authorized API actions I can help you to set up a personal bearer token. What is your use case?

**Comment by monotasker** [comment]
Ah, great to hear! Yes, we can definitely get the set up with you, but it's going to take a bit of development time to configure the oauth app authentication properly and I'm swamped just at the moment. (That's why the links to that settings page are hidden so far on KCWorks.) But go ahead and create a PAT for now and keep me posted on your plans. If you message hello@hcommons.org (or use the form here https://support.hcommons.org/contact-us/) we can put a ticket for you into the help system rather than trying to track progress here.

**Comment by monotasker** [comment]
(This issue was originally created, actually, for setting up additional scopes. But currently that's still on the to-do list.)

**Comment by mayel** [comment]
Ah thanks for the docs! I should have been looking at the InvenioRDM docs, as it seems the API differs from Zenodo's, no worries we'll just add another API adapter...

**Comment by monotasker** [comment]
No worries! Let me know if you have any other questions

**Comment by monotasker** [comment]
Moving to long-term planning doc
